flink-state-backends/flink-statebackend-changelog/src/main/java/org/apache/flink/state/changelog/ChangelogKeyedStateBackend.java

```diff
@@ -453,7 +465,10 @@ public void notifyCheckpointAborted(long checkpointId) throws Exception {
             // This might change if the log ownership changes (the method won't likely be needed).
             stateChangelogWriter.reset(lastUploadedFrom, lastUploadedTo);
         }
-        keyedStateBackend.notifyCheckpointAborted(checkpointId);
+        Long materializationID = materializationIdByCheckpointId.remove(checkpointId);
+        if (materializationID != null) {
```

Comment:
I do not think this logic is correct. Take materialization id as 1 for example, maybe chk-11 did not complete, but chk-12 could complete to involve the materialization-1. If we abort the materialization-1 at chk-11, RocksDB state backend would have to reupload at next materialization.

Reply:
I think you're right, although re-uploading will not do any harm.
I'll remove the abort call.

Reply:
Maybe removing the abort call directly is not good. We should better call this finally at appropriate moment.

Reply:
To summarize the offline discussion:
Pros of notifying: some backends might do GC
Cons of notifiying:
- there is low probability of abortion notification reaching backend (because of difference in intervals and cleanup on checkpoint completion)
- there is low probability of backends make good use of it (because it's delivered **after** snapshot is done; and backends must be resilient to missing notifications)
- there is added complexity and risk (please double check pre-marture notification and cleanup of maps)

I've added it in 862d28e7e2ff899a022d2f2e3da37ceaf149d31f but I'm still not sure whether it worth it.
PTAL.

Reply:
To summarize the offline discussion:
Pros of notifying: some backends might do GC
Cons of notifiying:
- there is low probability of abortion notification reaching backend (because of difference in intervals and cleanup on checkpoint completion)
- there is low probability of backends make good use of it (because it's delivered **after** snapshot is done; and backends must be resilient to missing notifications)
- there is added complexity and risk (please double check pre-marture notification and cleanup of maps)

I've added it in 862d28e7e2ff899a022d2f2e3da37ceaf149d31f but I'm still not sure whether it worth it.
PTAL.